go/libraries/doltcore/sqle/enginetest/dolt_transaction_queries.go

```diff
@@ -2370,4 +2380,322 @@ var MultiDbTransactionTests = []queries.ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "committing to another branch with autocommit",
+		SetUpScript: []string{
+			"create table t1 (a int)",
+			"call dolt_add('.')",
+			"call dolt_commit('-am', 'new table')",
+			"call dolt_branch('b1')",
+			"set autocommit = on", // unnecessary but make it explicit
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "insert into `mydb/b1`.t1 values (1)",
+				Expected: []sql.Row{
+					{types.OkResult{RowsAffected: 1}},
+				},
```

Comment:
If `@@dolt_transaction_commit` is enabled, does this commit on `mydb/b1` also create a dolt commit on that branch? Might be worth testing somewhere in here.